src/mpid/ch4/src/ch4_vci.h

```diff
@@ -70,6 +70,142 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_get_vci(int flag, MPIR_Comm * comm_ptr,
 }
 
 #elif MPIDI_CH4_VCI_METHOD == MPICH_VCI__IMPLICIT
+/* Remove the sign bit from an integer, if present */
+MPL_STATIC_INLINE_PREFIX int MPIDI_mask_int_sign(int i)
+{
+    return ((i) & ~(1 << (sizeof(int) * 8 - 1)));
+}
+
+/* Map comm to vci_idx */
+MPL_STATIC_INLINE_PREFIX int MPIDI_map_contextid_to_vci(MPIR_Context_id_t context_id)
+{
+    return (MPIDI_mask_int_sign(MPIR_CONTEXT_READ_FIELD(PREFIX, context_id))
+            % MPIDI_global.n_vcis);
+}
+
+/* Map comm and target_rank to vci_idx */
+MPL_STATIC_INLINE_PREFIX int MPIDI_map_contextid_rank_to_vci(MPIR_Context_id_t context_id,
+                                                             int target_rank)
+{
+    return (MPIDI_mask_int_sign(MPIR_CONTEXT_READ_FIELD(PREFIX, context_id) + target_rank)
+            % MPIDI_global.n_vcis);
+}
+
+/* Map comm and tag to vci_idx */
+MPL_STATIC_INLINE_PREFIX int MPIDI_map_contextid_tag_to_vci(MPIR_Context_id_t context_id, int tag)
+{
+    return (MPIDI_mask_int_sign(MPIR_CONTEXT_READ_FIELD(PREFIX, context_id) + tag)
+            % MPIDI_global.n_vcis);
+}
+
+/* Map comm, target_rank, and tag to vci_idx */
+MPL_STATIC_INLINE_PREFIX int MPIDI_map_contextid_rank_tag_to_vci(MPIR_Context_id_t context_id,
+                                                                 int target_rank, int tag)
+{
+    return (MPIDI_mask_int_sign(MPIR_CONTEXT_READ_FIELD(PREFIX, context_id) + target_rank + tag)
+            % MPIDI_global.n_vcis);
```

Comment:
I think you can remove `target_` and just say `rank`.

Reply:
Done